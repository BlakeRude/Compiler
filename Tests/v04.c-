outputs(char s[])
{
    int i;

    i = 0;
    while i<*s do {
          if s[i]=='\0' then break;
          outputc(s[i]);
          i++;
    }
}

main()
{
   char a[9]:"armadillo";
   char b[3]:"bat";
   char h[5]:"horse";
   char big[10];
   char small[2];

   small = a;
   outputs(small);
   outnl();
   big = a;
   outputs(big);
   outnl();
   outnl();

   small = b;
   outputs(small);
   outnl();
   big = b;
   outputs(big);
   outnl();
   outnl();

   small = h;
   outputs(small);
   outnl();
   big = h;
   outputs(big);
   outnl();
   outnl();
}   
