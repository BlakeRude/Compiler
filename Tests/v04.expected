Number of warnings: 0
Number of errors: 0
------------------------------------
Loading file: v04.tm
ar
armadillo
ba
batadillo
ho
horseillo
Bye.
